src/Libraries/RevitNodes/Elements/PathOfTravel.cs

```diff
@@ -0,0 +1,348 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Autodesk.DesignScript.Runtime;
+using Autodesk.Revit.DB;
+using Dynamo.Graph.Nodes;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+
+using Rvt = Autodesk.Revit.DB;
+using RvtAnalysis = Autodesk.Revit.DB.Analysis;
+
+namespace Revit.Elements
+{
+   /// <summary>
+   /// Revit Path of Travel Element
+   /// </summary>
+   [DynamoServices.RegisterForTrace]
+   public class PathOfTravel : Element
+   {
+      #region Internal properties
+
+      /// <summary>
+      /// An internal handle on the Revit path of travel
+      /// </summary>
+      RvtAnalysis.PathOfTravel m_rvtPathOfTravel = null;
+
+      /// <summary>
+      /// Reference to the Element
+      /// </summary>
+      [SupressImportIntoVM]
+      public override Rvt.Element InternalElement
+      {
+         get { return m_rvtPathOfTravel; }
+      }
+
+      #endregion
+
+      #region Private constructors
+
+      /// <summary>
+      /// PathOfTravel from existing
+      /// </summary>
+      /// <param name="rvtPathOfTravel">Revit PathOfTavel element</param>
+      private PathOfTravel(RvtAnalysis.PathOfTravel rvtPathOfTravel)
+      {
+         SafeInit(() => InitPathOfTravel(rvtPathOfTravel));
+      }
+
+      #endregion
+
+      #region Public constructors
+
+      /// <summary>
+      /// Construct a list of Path of Travel elements in a floor plan view between the specified start points and end points
+      /// </summary>
+      /// <param name="floorPlan">Floor plan view to place paths of travel on</param>
+      /// <param name="startPtList">List of start points</param>
+      /// <param name="endPtList">List of end points</param>
+      /// <param name="manyToMany">If true, paths are created from every point in the start point list to all points in the end point list. If false, a path is created from every point in the start point list to a corresponding point in the end point list with the same index. The two lists must have the same size when not creating many-to-many paths.</param>
+      /// <returns>List of Path of Travel elements; can contain null elements if there is no path between some points.</returns>
+      [NodeCategory("Create")]
+      [AllowRankReduction]
+      public static PathOfTravel[] ByFloorPlanPoints(Revit.Elements.Views.FloorPlanView floorPlan, Autodesk.DesignScript.Geometry.Point[] startPtList, Autodesk.DesignScript.Geometry.Point[] endPtList, bool manyToMany)
+      {
+         if (floorPlan == null)
+            throw new ArgumentNullException("Invalid view");
```

Comment:
these strings should get added to the .resx files in this same project so they can be localized by the localization team :)

Reply:
🤦‍♂